framework/src/meshgenerators/MeshGenerator.C

```diff
@@ -115,7 +198,27 @@ MeshGenerator::buildDistributedMesh(unsigned int dim)
 std::unique_ptr<MeshBase>
 MeshGenerator::generateInternal()
 {
+  libmesh_parallel_only(comm());
+  mooseAssert(comm().verify(type() + name()), "Inconsistent execution ordering");
+
+#ifndef NDEBUG
+  for (const auto & [name, mesh] : _requested_meshes)
+    mooseAssert(*mesh, "Null output from " + name);
+#endif
+
   auto mesh = generate();
+  mooseAssert(mesh, "Null output");
+
+  for (const auto & [requested_name, requested_mesh] : _requested_meshes)
+    if (*requested_mesh)
+      mooseError(
```

Comment:
I d consider a mooseWarning.

Anyone out there with MGs and poor unique pointer management would get an error now

Reply:
All apps pass and the error is descriptive and informative with a fix.

Laziness or misunderstanding is not a good reason to not practice proper memory management